File: me685/all codes/inverse.py
def inverse(m):
	n = len(m)
	q = unit(n)
	for i in range(n):
		m[i] = m[i] + q[i]

	I = [[0 for i in range(n)] for j in range(n)]
	
	for i in range(n):
		for j in range(n,2*n,1):
			if j-n == i:
				m[i][j] = 1
			else:
				m[i][j] = 0
	for i in range(n):
		x = i
		for j in range(i+1,n):
			y = abs(m[i][i])
			if y < abs(m[j][i]):
				x=j
				y = abs(m[j][i])
		for j in range(2*n):
			temp = m[i][j]
			m[i][j] = m[x][j]
			m[x][j] = temp
		temp = m[i][i]
		for k in range(2*n):
			m[i][k] = m[i][k]/temp
		for k in range(n):
			if k != i:
				p = m[k][i]
				for h in range(2*n):
					m[k][h] = m[k][h] - p*m[i][h]
	for i in range(n):
		for j in range(n):
			I[i][j] = m[i][n+j]
	return I


def unit(n):
	g = []
	for i in range(n):
		x = []
		for j in range(n-1,-1,-1):
			if i==j:
				x = x + [1]
			else:
				x = x + [0]
		g = [x] + g
	return g

File: ESO208/AS-3 - Copy/Inverse.m
function Inverse(A,n,maxit,relerr)
Y=ones(n,1);
count=0;
c=0;
for k=1:maxit
Y=A\Y;
a=max(abs(Y));
b=max(Y);
if(a==b)
    M=a;
else
    M=min(Y);
end
Y=rdivide(Y,M);
count=count+1;
e = 1/M;
if((e-c)*100/e<relerr)
    break;
end
c=e;
end
out=fopen('Inverse.txt','w');
fprintf(out,'Inverse power method\n');
fprintf(out,'Eigen value->%f\n',e);
fprintf(out,'Iterations->%f\n',count);
dlmwrite('Inverse.txt', Y/norm(Y),'-append');


end

